projects/libs/core-library-typescript/src/consensus/block-builder.ts

```diff
@@ -0,0 +1,36 @@
+import { types } from "../common-library";
+
+export default class BlockBuilder {
+    private virtualMachine: types.VirtualMachineClient;
+    private transactionPool: types.TransactionPoolClient;
+
+    constructor(input: {virtualMachine: types.VirtualMachineClient, transactionPool: types.TransactionPoolClient}) {
+        this.virtualMachine = input.virtualMachine;
+        this.transactionPool = input.transactionPool;
+    }
+
+    public async buildNextBlock(lastBlockId: number) {
+        const { transactions } = await this.transactionPool.getAllPendingTransactions({});
+
+        if (transactions.length == 0) {
+            throw "transaction pool is empty";
```

Comment:
Could you use here and below `throw new Error("transaction pool is empty");`) instead?